moveit_simple/src/robot.cpp

```diff
@@ -1319,8 +1367,14 @@ bool Robot::getIK(const Eigen::Isometry3d pose, const std::vector<double>& seed,
   {
     for (const std::pair<size_t, double> seed_state_fraction : ik_seed_state_fractions_)
     {
+      double midpoint;
       size_t joint = seed_state_fraction.first;
-      local_seed[joint] = local_seed[joint] * seed_state_fraction.second;
+      if (joint < ik_seed_state_mid_point_.size())
+        midpoint = ik_seed_state_mid_point_[joint];
+      else
+        midpoint = 0;
+
+      local_seed[joint] = local_seed[joint] * seed_state_fraction.second + midpoint * (1 - seed_state_fraction.second);
```

Comment:
Just a bit shorter, but your version is also good
```suggestion
      local_seed[joint] -= ( local_seed[joint] - midpoint ) * seed_state_fraction.second;
```

Reply:
yup. but It works and I like the explicit weighted average.